TestApp/src/main/java/com/tekchand/testapp/ui/main/vmfrag.java

```diff
@@ -0,0 +1,158 @@
+package com.tekchand.testapp.ui.main;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.gson.Gson;
+import com.tekchand.testapp.R;
+import com.tekchand.testapp.ui.main.pojos.ItemsItem;
+import com.tekchand.testapp.ui.main.pojos.Responses;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class vmfrag extends Fragment {
+
+    private VmfragViewModel mViewModel;
+    private Gson gson = new Gson();
+    private RecyclerView recyclerView;
+    private OnVideoFragmentListener mListener;
+    private static List<Video> videos = new ArrayList<>();
+
+
+    public static vmfrag newInstance() {
+        return new vmfrag();
+    }
+
+    private String apiKey = "AIzaSyCjmYJ7OKn0CQtxByFBxYkwGVU8jb7hV7k";
```

Comment:
API keys should be stored in Gradle as a string resource. This will be convenient when the application has multiple build variants for different environments.